src/components/TodoForm.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from 'react';
+import usersFromServer from '../api/users';
+
+interface TodoFormProps {
+  onAddTodo: (title: string, userId: number) => void;
+}
+
+export const TodoForm: React.FC<TodoFormProps> = ({ onAddTodo }) => {
+  const [userSelectId, setUserSelectId] = useState(0);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [errorUserSelectId, setErrorUserSelectId] = useState(false);
+  const [errorTodoTitle, setErrorTodoTitle] = useState(false);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (userSelectId === 0) {
+      setErrorUserSelectId(true);
+    }
+
+    if (todoTitle === '') {
+      setErrorTodoTitle(true);
+    }
+
+    if (userSelectId === 0 || todoTitle === '') {
+      return;
+    }
+
+    onAddTodo(todoTitle, userSelectId);
+    setUserSelectId(0);
+    setTodoTitle('');
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value.replace(/[^a-zA-Z0-9 ]/g, '');
```

Comment:
The regex used in `handleTitleChange` to sanitize input only allows alphanumeric characters and spaces. This might be too restrictive if you want to allow other characters like punctuation. Consider adjusting the regex to allow a broader range of characters if needed.